### DISCUSSION TITLE
Is it possible to change a standalone deployment to a cluster deployment?

### DISCUSSION THREAD

**Discussion by pycui** [milvus-io/milvus#35888] [Q&A and General discussion]
If yes, what are the recommended procedure? Thank you.

**Comment by xiaofan-luan** [comment]
if you are using standalone with kafka and external etcd, then it's simple, you can simply change the config and reboot.

If you use standalone with rocksmq the you need some backup and restore in a new cluster i guess

**Reply by pycui** [reply]
Thanks for the reply. I'm using the default message queue which I believe is rocksmq. Can you share some guide on how to properly do the backup and restore in a new cluster? thank you

**Comment by xiaofan-luan** [comment]
https://github.com/zilliztech/milvus-backup this is the backup tool we offered

**Reply by pycui** [reply]
Cool. So I will set source minio of the standalone deployment, and backup minio using the new cluster deployment. Apart from this, is there anything else to restore in the new cluster? Just want to be sure. Thanks

**Reply by yhmo** [reply]
> Cool. So I will set source minio of the standalone deployment, and backup minio using the new cluster deployment. Apart from this, is there anything else to restore in the new cluster? Just want to be sure. Thanks

https://milvus.io/docs/milvus_backup_overview.md
https://milvus.io/docs/milvus_backup_cli.md
The milvus-backup is an independent server that accepts cli commands or restful requests to do backup&restore works.

For backup command:
1. it gets the collection schema from Etcd of the source milvus
2. it stores the schema as a xml/yaml file into a backup path of a S3
3. it copies the data files of the collection from S3 of the source milvus.
4. it stores the data files into a backup path of a S3

For restore command:
1. it reads the collection schema from the backup path, call the interface of the target milvus to create a new collection
2. it calls the bulkinsert interface of the target milvus to import the data files from the backup path.

To execute the two commands, the backup.yaml needs to be configured carefully.
Here is a document for how to use this tool in different cases:
[milvus-backup tool practice.pdf](https://github.com/user-attachments/files/16832263/milvus-backup.tool.practice.pdf)
